packages/protocol/engine/ledger/mempool/conflictdag/conflictdag.go

```diff
@@ -24,6 +25,8 @@ type ConflictDAG[ConflictIDType, ResourceIDType comparable] struct {
 	// mutex is a mutex that prevents that two processes simultaneously update the ConflictDAG.
 	mutex *syncutils.StarvingMutex
 
+	WeightsMutex sync.RWMutex
```

Comment:
This mutex is very weird. It says it is locking access to weights but is not used in this pkg but locked/unlocked from the outside.

Reply:
Yes, it will be refactored along with the whole ConflictDAG/ConflictTracker/ConflictResolver rework.